### PR TITLE
Change JavaGenerator to generate abstract adapters.

### PR DESCRIPTION
**Pull request by swankjesse**
Previously we generated small classes that described the fields of the
proto. This was awkward to use.

### CODE REVIEW

File: wire-maven-plugin/src/test/resources/exemplar/src/main/java/com/squareup/wire/exemplar/Exemplar.java [hunk 1]
```diff
@@ -1,48 +1,24 @@
...
+import com.squareup.protos.wire.exemplar.AbstractLocaleAdapter;
 import com.squareup.protos.wire.exemplar.MuchAwesome;
-import com.squareup.wire.FieldEncoding;
 import com.squareup.wire.ProtoAdapter;
-import com.squareup.wire.ProtoReader;
-import com.squareup.wire.ProtoWriter;
 import java.io.IOException;
 import java.util.Locale;
 
 public class Exemplar {
```
Comment (swankjesse) [inline wire-maven-plugin/src/test/resources/exemplar/src/main/java/com/squareup/wire/exemplar/Exemplar.java]: Before/After is here.

File: wire-java-generator/src/main/java/com/squareup/wire/java/JavaGenerator.java [hunk 2]
```diff
@@ -740,22 +738,21 @@ private MethodSpec messageAdapterEncodedSize(NameAllocator nameAllocator, Messag
...
       CodeBlock adapter = adapterFor(field);
-      if (field.isRequired() || field.isRepeated() || field.type().isMap()) {
-        result.addCode("$L $L.encodedSizeWithTag($L, value.$L)",
-            leading, adapter, fieldTag, fieldName);
-      } else {
-        result.addCode("$1L (value.$4L != null ? $2L.encodedSizeWithTag($3L, value.$4L) : 0)",
-            leading, adapter, fieldTag, fieldName);
-      }
+      result.addCode("$L $L.encodedSizeWithTag($L, ", leading, adapter, fieldTag)
+          .addCode((useBuilder ? "value.$L" : "$L(value)"), fieldName)
```
Comment (JakeWharton) [inline wire-java-generator/src/main/java/com/squareup/wire/java/JavaGenerator.java]: nit: extra parens

File: wire-java-generator/src/main/java/com/squareup/wire/java/JavaGenerator.java [hunk 3]
```diff
@@ -765,92 +762,135 @@ private MethodSpec messageAdapterEncode(NameAllocator nameAllocator, MessageType
...
+      if (field.isRedacted()) {
+        redactedFieldCount++;
+        if (field.isRequired()) {
+          requiredRedacted.add(nameAllocator.get(field));
+        }
       }
     }
+
+    if (!useBuilder) {
+      result.addStatement((redactedFieldCount == 0) ? "return value" : "return null");
```
Comment (JakeWharton) [inline wire-java-generator/src/main/java/com/squareup/wire/java/JavaGenerator.java]: nit: extra parens

**Comment by JakeWharton** [comment]
It's more beautiful than I imagined 😂

**Pull request merged**